Source/Tools/FEXLoader/FEXLoader.cpp

```diff
@@ -194,15 +193,53 @@ void RootFSRedirect(fextl::string* Filename, const fextl::string& RootFS) {
   }
 }
 
-bool RanAsInterpreter(const char* Program) {
-  return ExecutedWithFD || FEXLOADER_AS_INTERPRETER;
+FEX::Config::PortableInformation ReadPortabilityInformation() {
+  const FEX::Config::PortableInformation BadResult {false, {}};
+  const char* PortableConfig = getenv("FEX_PORTABLE");
+  if (!PortableConfig) {
+    return BadResult;
+  }
+
+  uint32_t Value {};
+  std::string_view PortableView {PortableConfig};
+
+  if (std::from_chars(PortableView.data(), PortableView.data() + PortableView.size(), Value).ec != std::errc {} || Value == 0) {
+    return BadResult;
+  }
+
+  // Read the FEXInterpreter path from `/proc/self/exe` which is always a symlink to the absolute path of the executable running.
+  // This way we can get the parent path that the application is executing from.
+  char SelfPath[PATH_MAX];
+  auto Result = readlink("/proc/self/exe", SelfPath, PATH_MAX);
+  if (Result == -1) {
+    return BadResult;
+  }
+
+  std::string_view SelfPathView {SelfPath, std::min<size_t>(PATH_MAX, Result)};
+
+  // Extract the absolute path from the FEXInterpreter path
+  return {true, fextl::string {SelfPathView.substr(0, SelfPathView.find_last_of('/') + 1)}};
 }
 
-bool IsInterpreterInstalled() {
-  // The interpreter is installed if both the binfmt_misc handlers are available
-  // Or if we were originally executed with FD. Which means the interpreter is installed
+bool RanAsInterpreter(bool ExecutedWithFD) {
+  return ExecutedWithFD || FEXLOADER_AS_INTERPRETER;
+}
 
-  return ExecutedWithFD || (access("/proc/sys/fs/binfmt_misc/FEX-x86", F_OK) == 0 && access("/proc/sys/fs/binfmt_misc/FEX-x86_64", F_OK) == 0);
+/**
+ * @brief Queries if FEX is installed as a binfmt_misc interpreter
+ *
+ * @param ExecutedWithFD If FEXInterpreter was executed using a binfmt_misc FD handle from the kernel
+ * @param Portable Portability information about FEX being run in portable mode
+ *
+ * @return true if the binfmt_misc handlers are installed and being used
+ */
+bool QueryInterpreterInstalled(const bool ExecutedWithFD, const FEX::Config::PortableInformation& Portable) {
+  // We know the FEX interpreter is installed through a couple paths.
```

Comment:
Unclear what this line is trying to say. ~~Are we *checking* any paths here to figure out if the FEX interpreter is installed? Which paths? None of the bullet points after mention any paths.~~

On third read, it seems you're using "path" in a figurative way, which is confusing in code that also does filesystem operations... Is the intent here that we have multiple *methods* of checking whether FEXInterpreter is installed?

Reply:
Yes, if the paths exist, or if `ExecutedWithFD` then we know that the Linux kernel binfmt_misc interpreter is installed. `/proc` doesn't need to be mounted inside of a chroot for example, so those two paths checks could not exist, but binfmt_misc interpreter is still installed.

Reply:
This is probably clearer:

```
  // Check if FEX's binfmt_misc handlers are both installed.
  // The explicit check can be omitted if FEX was executed from an FD,
  // since this only happens if the kernel launched FEX through binfmt_misc
```

Reply:
Actually, scratch that, let me revisit this.

Reply:
I guess my proposal captures the essence of it, though it would be nice if the nuance of /proc possibly not being visible could be factored in as well. Then again `/proc` is *typically* mounted in a chroot, so this may be too much of an edge case to bother going into the detail.

Anyway, seems good enough.